src/lib/UtapiClient.js

```diff
@@ -329,40 +328,61 @@ export default class UtapiClient {
+        const paramsArr = this._getParamsArr(params);
+        paramsArr.forEach(p => {
+            cmds.push(
+                ['incrby', generateCounter(p, 'storageUtilizedCounter'),
+                    newByteLength],
+                ['incrby', generateKey(p, 'incomingBytes', timestamp),
+                    newByteLength],
+                ['incr', generateKey(p, action, timestamp)]
+            );
+        });
         // update counters
-        return this.ds.batch([
-            ['incrby', generateCounter(params, 'storageUtilizedCounter'),
-                newByteLength],
-            ['incrby', generateKey(params, 'incomingBytes', timestamp),
-                newByteLength],
-            ['incr', generateKey(params, action, timestamp)],
-        ], (err, results) => {
+        return this.ds.batch(cmds, (err, results) => {
             if (err) {
                 log.error('error pushing metric', {
                     method: 'UtapiClient._pushMetricUploadPart',
                     error: err,
                 });
                 return callback(errors.InternalError);
             }
-            // storage utilized counter
-            const actionErr = results[0][0];
-            const actionCounter = results[0][1];
-            if (actionErr) {
-                log.error('error incrementing counter for push metric', {
-                    method: 'UtapiClient._pushMetricUploadPart',
-                    metric: 'storage utilized',
-                    error: actionErr,
-                });
-                return callback(errors.InternalError);
+            // storage utilized counters
+            let index;
+            let actionErr;
+            let actionCounter;
+            const cmdsLen = cmds.length;
+            const paramsArrLen = paramsArr.length;
+            const cmds2 = [];
+            const noErr = paramsArr.every((p, i) => {
+                // index corresponds to the result from the previous set of
+                // commands. we are trying to extract the storage utlized
+                // counters per metric level here.
+                index = i * (cmdsLen / paramsArrLen);
+                actionErr = results[index][0];
+                actionCounter = results[index][1];
+                if (actionErr) {
+                    log.error('error incrementing counter for push metric', {
+                        method: 'UtapiClient._pushMetricUploadPart',
+                        metric: 'storage utilized',
+                        error: actionErr,
+                    });
+                    callback(errors.InternalError);
+                    return false;
+                }
+                cmds2.push(
+                    ['zremrangebyscore', generateStateKey(p, 'storageUtilized'),
+                        timestamp, timestamp],
+                    ['zadd', generateStateKey(p, 'storageUtilized'),
+                        timestamp, actionCounter]
+                );
+                return true;
```

Comment:
if you do this multiple times I feel like you can probably refactor this every check into a util function, e.g. you can pass cmds.length, paramsArr.length, logger, and the types of counters you want to update